### DISCUSSION TITLE
Install dependencies of a package and make a `view` of them but do not install the package itself?

### DISCUSSION THREAD

**Discussion by brettviren** [spack/spack#44446] [General]
tl;dr: Is there a way to install only the dependencies of a package and build a spack view from them without actually installing the package itself?

Details:

I have this `package.py` provided to Spack with a user repo.

https://github.com/WireCell/wire-cell-spack/blob/master/packages/wire-cell-toolkit/package.py

I like to use `spack view` to provide a development environment for working on wire-cell-toolkit source itself.   But, in those cases, I only want the dependencies and do not want Spack to actually build our software or have any of its bits linked into a `view`.

I have tried these approaches but none are quite optimal.

* Have spack do the install of the package and build the view based on it but then `rm` the files (symlinks) from the view provided by our package (eg so as not to confuse the development build).  This works but is tedious and easy to forget when making a new view.  There are also times when updating Spack causes build problems with our software (eg, lately dealing with fmtlib v9 API changes) so I can't even exercise this method.

* Maintain a separate `BundlePackage` (`wire-cell-dependencies`) which consists of copy-pasted code from the bulk of `wire-cell-toolkit/package.py`.  This works but it is tedious to maintain and it is easy to lose sync between the bundle and the "real" `package.py` as updates are made.

* Make the real package "empty" but depend on the bundle.  In principle this also works but requires a lot of verbosity to "forward" variants and versions via `depends_on()`.  It also requires inventing bundle versions as dependencies change which would otherwise be implicit when everything lumped in the "real" package.

* Add a variant `depsonly` that causes the `install()` method to simply return.  Spack doesn't like this as it detects the non-installation as an error.  I guess I could "install" by touching a dummy file but it feels too hacky.

Is there some way to have my cake and eat it too here?

**Comment by brettviren** [comment]
I should note there is 

```
spack view -e <seed> add -i local <seed>
```

Which will exclude `<seed>` from the view, but still requires it to be installed.

**Comment by psakievich** [comment]
Have you tried `spack install --only dependencies && spack view`?  Does that work?

**Comment by brettviren** [comment]
Yes, I'm pretty sure I tried that.  The `install` worked but then the `view -e <excluded>` wants the `<excluded>` package to be installed.

**Reply by psakievich** [reply]
You shouldn't need the exclude though if it wasn't installed? I can try to test it out this week.

**Reply by brettviren** [reply]
It does not seem to work.

For a test, I pick a low-dependency package `aspell` that happens to not be currently installed in my area:

```
$ spack find aspell
==> Error: No package matches the query: aspell

$ spack install --only dependencies aspell
[+] /usr (external glibc-2.36-bvtf6bex4zbdpjjoeglekzlulmcgwycu)
[+] /wcwc/spack/local/opt/spack/linux-debian12-haswell/gcc-13.2.0/gcc-runtime-13.2.0-w2gkrm6xgey4hc3dks6idlziun25dkfi
[+] /wcwc/spack/local/opt/spack/linux-debian12-haswell/gcc-12.2.0/gcc-runtime-12.2.0-rzcsqo6q3sdwvp4estdzenmnm6zwkqow
[+] /wcwc/spack/local/opt/spack/linux-debian12-haswell/gcc-12.2.0/gmake-4.4.1-7mvfiwagdjzskvmdxr6uoljzohgplo2k

$ spack view add -i /tmp/not-aspell aspell
==> Error: Spec 'aspell' matches no installed packages.

$ spack view -e aspell add -i /tmp/not-aspell aspell
==> Error: Spec 'aspell' matches no installed packages.
```
